tests/unittests/InterpreterTest.cpp

```diff
@@ -898,6 +898,90 @@ TEST(Interpreter, nonLinearClassifier) {
   }
 }
 
+/// Generate images in two classes. A "line" is labeled as 0
+/// and a "cross" is labeled as 1
+static void generateImageData(Tensor &images, Tensor &labels, unsigned numSamples, unsigned isTrain) {
+  auto L = labels.getHandle<size_t>();
+  auto image = images.getHandle<>();
+  images.zero();
+  
+  for (size_t i = 0; i < numSamples; i++) {
+    bool isLine = i % 2 == 0;
+    L.at({i, 0}) = isLine ? 0 : 1;
+    size_t target;
+    // to make sure testing set is not totally included in 
+    // training set
+    if (isTrain)  {
+      target = rand() % 25 + 1;
+    }
+    else {
+      target = rand() % 30 + 1;
+    }
+    for (size_t x = 0; x < 32; x++) {
+      for (size_t y = 0; y < 32; y++) {
+        if (isLine) {
+	  // line
+	  if (x == target)
+            image.at({i, x, y, 0u}) = 1;
+        } else {
+          // cross
+          if (x == target || y == target)
+            image.at({i, x, y, 0u}) = 1;
+        }
+      }
+    }
+  }
+}
+
+TEST(Interpreter, simpleConv) {
+  unsigned numSamples = 200;
+  ExecutionEngine EE;
+
+  EE.getConfig().learningRate = 0.01;
+  EE.getConfig().batchSize = numSamples;
+  EE.getConfig().momentum = 0.9;
+
+  auto &mod = EE.getModule();
+  Function *F = mod.createFunction("main");
+
+  auto *input = mod.createVariable(ElemKind::FloatTy, {numSamples, 32, 32, 1}, "input",
+                                   VisibilityKind::Public, Variable::TrainKind::None);
+
+  auto *ex = mod.createVariable(ElemKind::IndexTy, {numSamples, 1}, "exp",
+                                VisibilityKind::Public, Variable::TrainKind::None);
+
+  auto *CV0 = F->createConv("conv1", input, 1, 3, 1, 0, 1);
+  auto *RL0 = F->createRELU("relu1", CV0);
+  auto *FCL1 = F->createFullyConnected("fc", RL0, 2);
+  auto *SM = F->createSoftMax("sm", FCL1, ex);
+  auto *result = F->createSave("ret", SM);
+  Function *TF = glow::differentiate(F, EE.getConfig());
+  EE.compile(CompilationMode::Train, TF);
+
+  Tensor images(ElemKind::FloatTy, {numSamples, 32, 32, 1});
+  Tensor labels(ElemKind::IndexTy, {numSamples, 1});
+  generateImageData(images, labels, numSamples, 1);
+
+  // Training:
+  EE.runBatch(1000, {input, ex}, {&images, &labels});
+  EE.compile(CompilationMode::Infer, F);
+
+  // generate the images used for testing.
+  Tensor testImage(ElemKind::FloatTy, {numSamples, 32, 32, 1});
+  generateImageData(testImage, labels, numSamples, 0);
+  EE.run({input}, {&testImage});
+  int success = 0;
+  auto SMH = result->getVariable()->getHandle<>();
+  for (size_t i = 0; i < numSamples; i++) {
+    bool isLine = (i % 2 == 0);
+    auto A = SMH.at({i, 0});
+    auto B = SMH.at({i, 1});
+    if ((isLine && A > B) || (!isLine && B > A))
+      success++;
+  }
+  EXPECT_TRUE((float)success / numSamples > 0.85);
```

Comment:
what's the current reported success rate?

Reply:
might just use, EXPECT_GT

Reply:
85% is a bit close to 88.5%, so I think the test might be a bit fragile based on the random numbers provided during training/testing. Maybe use 80% or something? Have you tried using different random number seeds, or looping on the test many times like we discussed before to make sure the test doesn't fail due to accuracy below 85%?